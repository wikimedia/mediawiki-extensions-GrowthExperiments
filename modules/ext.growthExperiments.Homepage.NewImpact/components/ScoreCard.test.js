const { mount } = require( '@vue/test-utils' );
const ScoreCard = require( './ScoreCard.vue' );

describe( 'ScoreCard', () => {
	it( 'renders correctly', () => {
		const wrapper = mount( ScoreCard, {
			props: {
				icon: 'some icon',
				iconLabel: 'some icon label',
				label: 'some label'
			},
			global: {
				provide: {
					RENDER_MODE: 'desktop'
				}
			},
			slots: {
				default: '123'
			}
		} );
		expect( wrapper.element ).toMatchSnapshot();
	} );
	it( 'renders correctly info slot', () => {
		const wrapper = mount( ScoreCard, {
			props: {
				icon: 'some icon',
				iconLabel: 'some icon label',
				label: 'some label'
			},
			global: {
				provide: {
					RENDER_MODE: 'desktop'
				}
			},
			slots: {
				default: '123',
				'label-info': '<p>Some info text</p>'
			}
		} );
		expect( wrapper.element ).toMatchSnapshot();
	} );
} );
