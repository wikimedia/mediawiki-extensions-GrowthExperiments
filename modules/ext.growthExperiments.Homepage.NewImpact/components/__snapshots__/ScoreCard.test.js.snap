// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreCard renders correctly 1`] = `
<div
  class="ext-growthExperiments-ScoreCard"
>
  <div
    class="ext-growthExperiments-ScoreCard__data-display"
  >
    <span
      class="cdx-icon ext-growthExperiments-ScoreCard__data-display__icon"
    >
      <svg
        aria-hidden="false"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          some icon label
        </title>
        <g
          fill="currentColor"
        >
          some icon
        </g>
      </svg>
    </span>
    
    123
    
  </div>
  <div
    class="ext-growthExperiments-ScoreCard__label"
  >
    <span
      class="ext-growthExperiments-CText ext-growthExperiments-CText--color-subtle ext-growthExperiments-ScoreCard__label__text"
    >
      
      some label
      
    </span>
    
    
  </div>
</div>
`;

exports[`ScoreCard renders correctly info slot 1`] = `
<div
  class="ext-growthExperiments-ScoreCard"
>
  <div
    class="ext-growthExperiments-ScoreCard__data-display"
  >
    <span
      class="cdx-icon ext-growthExperiments-ScoreCard__data-display__icon"
    >
      <svg
        aria-hidden="false"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          some icon label
        </title>
        <g
          fill="currentColor"
        >
          some icon
        </g>
      </svg>
    </span>
    
    123
    
  </div>
  <div
    class="ext-growthExperiments-ScoreCard__label ext-growthExperiments-ScoreCard__label--with-info"
  >
    <span
      class="ext-growthExperiments-CText ext-growthExperiments-CText--color-subtle ext-growthExperiments-ScoreCard__label__text"
    >
      
      some label
      
    </span>
    
    <p>
      Some info text
    </p>
    
  </div>
</div>
`;
